javav2/example_code/pinpoint/src/main/java/com/example/pinpoint/ExportEndpoints.java

```diff
@@ -0,0 +1,228 @@
+                .build();
+
+        S3Client s3Client = S3Client.builder()
+                .region(region)
+                .build();
+
+        exportAllEndpoints(pinpoint, s3Client, applicationId, s3BucketName, path, iamExportRoleArn);
+        pinpoint.close();
+        s3Client.close();
+   }
+
+    //snippet-start:[pinpoint.java2.export_endpoint.main]
+    public static void exportAllEndpoints(PinpointClient pinpoint,
+                                          S3Client s3Client,
+                                          String applicationId,
+                                          String s3BucketName,
+                                          String path,
+                                          String iamExportRoleArn) {
+
+        try {
+
+            List<String> objectKeys = exportEndpointsToS3(pinpoint, s3Client, s3BucketName, iamExportRoleArn, applicationId);
+            List<String> endpointFileKeys = objectKeys.stream().filter(o -> o.endsWith(".gz")).collect(Collectors.toList());
+            downloadFromS3(s3Client, path, s3BucketName, endpointFileKeys);
+
+        } catch ( PinpointException e) {
+            System.err.println(e.awsErrorDetails().errorMessage());
+            System.exit(1);
+        }
+  }
+
+    public static List<String> exportEndpointsToS3(PinpointClient pinpoint, S3Client s3Client, String s3BucketName, String iamExportRoleArn, String applicationId) {
+
+        SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd-HH_mm:ss.SSS_z");
+        String endpointsKeyPrefix = "exports/" + applicationId + "_" + dateFormat.format(new Date());
+        String s3UrlPrefix = "s3://" + s3BucketName + "/" + endpointsKeyPrefix + "/";
+        List<String> objectKeys = new ArrayList<>();
+        String key ="" ;
+
+        try {
+            // Defines the export job that Amazon Pinpoint runs
+            ExportJobRequest jobRequest = ExportJobRequest.builder()
+                    .roleArn(iamExportRoleArn)
+                    .s3UrlPrefix(s3UrlPrefix)
+                    .build();
+
+            CreateExportJobRequest exportJobRequest =  CreateExportJobRequest.builder()
+                    .applicationId(applicationId)
+                    .exportJobRequest(jobRequest)
+                    .build();
+
+            System.out.format("Exporting endpoints from Amazon Pinpoint application %s to Amazon S3 " +
+                    "bucket %s . . .\n", applicationId, s3BucketName);
+
+            CreateExportJobResponse exportResult = pinpoint.createExportJob(exportJobRequest);
+            String jobId = exportResult.exportJobResponse().id();
+            System.out.println(jobId);
+            printExportJobStatus(pinpoint, applicationId, jobId);
+
+            ListObjectsV2Request v2Request = ListObjectsV2Request.builder()
+                    .bucket(s3BucketName)
+                    .prefix(endpointsKeyPrefix)
+                    .build();
+
+            // Create a List of Object keys

```

Comment:
"Object" -> "object" (not a proper name)

Reply:
fixed

Reply:
fixed